sympy/sets/fancysets.py

```diff
@@ -1443,8 +1441,13 @@ def _union(self, other):
             elif self.polar and other.polar:
                 return ComplexRegion(Union(self.sets, other.sets), polar=True)
 
-        if self == S.Complexes:
-            return self
+        # treat a subset of Reals as complex numbers
+        if other.is_subset(S.Reals):
```

Comment:
The method could probably check this first and coerce other into a ComplexRegion before asking `if other.is_ComplexRegion`.